.maintain/gitlab/check_companion_build.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env sh
+#
+# check if a pr is compatible with companion pr or master if not available
+#
+# to override one that was just mentioned mark companion pr in the body of the
+# pr like
+#
+# $REPO companion: $ORGANISATION/$REPO#567
+#
+# $ORGANISATION and $REPO are set using $1 and $2. You can also specify a custom
+# build command with $3
+# So invoke this script like:
+# ./check_companion_build.sh paritytech polkadot 'cargo test --release'
+
+set -e
+
+ORGANISATION=$1
+REPO=$2
+BUILDSTRING=${3:-cargo test --all --release}
+
+github_api_substrate_pull_url="https://api.github.com/repos/paritytech/substrate/pulls"
+# use github api v3 in order to access the data without authentication
+github_header="Authorization: token ${GITHUB_PR_TOKEN}"
+
+boldprint () { printf "|\n| \033[1m%s\033[0m\n|\n" "${@}"; }
+boldcat () { printf "|\n"; while read -r l; do printf "| \033[1m%s\033[0m\n" "${l}"; done; printf "|\n" ; }
+
+
+
+boldcat <<-EOT
+
+
+check_${REPO}_companion_build
+==============================
+
+this job checks if there is a string in the description of the pr like
+
+$REPO companion: $ORGANISATION/$REPO#567
+
+
+it will then run cargo check from this ${REPO}'s branch with substrate code
+from this pull request. otherwise, it will uses master instead
+
+
+EOT
+
+# Set the user name and email to make merging work
+git config --global user.name 'CI system'
+git config --global user.email '<>'
+
+# Merge master into our branch before building to make sure we don't miss
+# any commits that are required.
+git fetch --depth 100 origin
+git merge origin/master
+
+# Clone the current master branch into a local directory
+# NOTE: we need to pull enough commits to be able to find a common
+# ancestor for successfully performing merges below.
+git clone --depth 20 "https://github.com/${ORGANISATION}/${REPO}.git"
+
+cd "$REPO"
+
+# either it's a pull request then check for a companion otherwise use
+# master
+# shellcheck disable=SC2003
+if expr match "${CI_COMMIT_REF_NAME}" '^[0-9]\+$' >/dev/null
+then
+  boldprint "this is pull request no ${CI_COMMIT_REF_NAME}"
+
+  pr_data_file="$(mktemp)"
+  # get the last reference to a pr in the target repo
+  curl -sSL -H "${github_header}" -o "${pr_data_file}" \
+    "${github_api_substrate_pull_url}/${CI_COMMIT_REF_NAME}"
+
+  pr_body="$(sed -n -r 's/^[[:space:]]+"body": (".*")[^"]+$/\1/p' "${pr_data_file}")"
+
+  pr_companion="$(echo "${pr_body}" | sed -n -r \
```

Comment:
Twice the companion merge was blocked due to https://github.com/isaacs/github/issues/1681

- https://github.com/paritytech/polkadot/pull/3086
- https://github.com/paritytech/polkadot/pull/2922

Is it feasible to fail the check if the companion PR is coming from an organization repository? It's possible to query that using the Github API, e.g. https://api.github.com/users/paritytech will return `"type": "Organization"`

Suggested message: `Heads-up: You've opened the companion PR from an organization repository; bear in mind "bot merge" will not also merge the companion simultaneously due to a Github limitation (link).`

Reply:
yeah, for sure let's add this warning with mentioning it's temporary until that bug is addressed.